ciao-controller/compute.go

```diff
@@ -906,7 +911,16 @@ func createServer(w http.ResponseWriter, r *http.Request, context *controller) {
 	}
 	servers.TotalServers = len(instances)
 
-	b, err := json.Marshal(servers)
+	// set machine ID for OpenStack compatibility
+	server.Server.ID = instances[0].ID
```

Comment:
@markdryan right, startWorkload verify the number of instances is not 0, so that scenario is covered :)  https://github.com/01org/ciao/blob/master/ciao-controller/command.go#L127